erpnext/payroll/doctype/salary_slip/salary_slip.py

```diff
@@ -107,7 +107,7 @@ def check_existing(self):
 		if not self.salary_slip_based_on_timesheet:
 			ret_exist = frappe.db.sql("""select name from `tabSalary Slip`
 						where start_date = %s and end_date = %s and docstatus != 2
-						and employee = %s and name != %s""",
+						and employee = %s and name != %s and payroll_entry='' """,
```

Comment:
the duplicate should be checked irrespective of it is created against payroll_entry or not.